dashboard/app/models/script.rb

```diff
@@ -1472,6 +1472,8 @@ def get_assessment_script_levels
   def get_feedback_for_section(section)
     feedback = {}
     script_levels.each do |script_level|
+      # Filter down to only the levels a teacher can give feedback on
+      next unless ["Applab", "Gamelab", "Weblab"].include?(script_level.level.type)
       section.students.each do |student|
         temp_feedback = TeacherFeedback.get_student_level_feedback(student.id, script_level.level.id, section.user_id)
```

Comment:
If you're having performance problems on this route, it's probably because there are nested database queries.  I suspect it'd be faster to do a few large queries and do any additional grouping in-memory afterwards.  Something like:

```rb
level_ids = script_levels.map(&:level).select(&:can_have_feedback?).map(&:id)
student_ids = section.students.map(&:id)
feedbacks = TeacherFeedback.get_students_levels_feedback(student_ids, level_ids, section.user_id)
# Transform into desired hash...
```
Where we've set up `get_students_levels_feedback` to retrieve all the latest feedback for all students x levels in one query.

Reply:
i'm looking into this option as well. the only part i'm having trouble with is that we need to maintain a reference to `script_level` so that we can create our feedback object with it (lines 1482-1484). 

i think we need to join feedback to script_levels on level.id somehow (continuing your example from above):
```
feedbacks.joins("INNER JOIN script_levels ON script_levels.level_id = feedbacks.level_id")
```
except `script_levels.level_id` doesn't exist (script_level has_and_belongs_to_many :levels), so i'm a little stuck

Reply:
@dmcavoy just to double check, can we remove `stageNum`, `stageName`, and `levelNum` from our feedback object? that will simplify this query a lot, so i want to step back to evaluate if we can do without that data in this context

Reply:
Whoops! That points out a bug in this PR.

Reply:
Hmm, good question.  `feedback.level.script_levels` isn't quite right because it's not limited to this script.  I think, having the script_levels and levels you care about in memory already to build the query, there's still some way to do this with only a few round-trips to the db.

Reply:
@madelynkasula I don't think we can go without those because thats what tells you the lesson name and number as well as the level number which we need for the CSV.